### DISCUSSION TITLE
TypeError: Cannot read properties of null (reading 'document') with App Router and Intercom

### DISCUSSION THREAD

**Discussion by kendis-fit** [vercel/next.js#67041] [Help] [answered]
### Summary

I need to boot/show intercom when users move between pages

### Additional information

Intercom is shown for a first page, but the error mentioned in title occurs if users try to move to another page.
The object Intercom persists between pages, but its methods don't work, if I try to use the method boot, then as I mentioned I got the error, if I try to use method update, then nothing happens.

Intercom provides simple page app solution as a script, but it seems it doesn't work with nextjs applications, because I have one more project which is written in React, and it doesn't have the problem, Intercom's UI persists between pages.

**Comment by ahmedreza1** [comment]
@kendis-fit were you able to fix it?

**Reply by kendis-fit** [reply, marked as answer]
Yes, I've talked to Intercom support who had provided a script which I've used in my component, take a look
use it for every page where you need it
```
'use client';

import { useEffect } from 'react';

const appId = YOUR_TOKEN;

const IntercomScript = () => {
  useEffect(() => {
    let d = document;
    let i: any = function () {
      i.c(arguments);
    };
    i.q = [];
    i.c = function (args: any) {
      i.q.push(args);
    };
    window.intercomSettings = {
      app_id: appId,
    }
    window.Intercom = i;
    let l = function () {
      let _a, _b;
      let s = d.createElement('script');
      s.type = 'text/javascript';
      s.async = true;
      s.src = `https://widget.intercom.io/widget/${((_a = window.intercomSettings) === null || _a === void 0 ? void 0 : _a.app_id)}`;
      let x = d.getElementsByTagName('script')[0];
      (_b = x.parentNode) === null || _b === void 0 ? void 0 : _b.insertBefore(s, x);
    };
    if (document.readyState === 'complete') {
      l();
    }
    else if (window.attachEvent) {
      window.attachEvent('onload', l);
    }
    else {
      window.addEventListener('load', l, false);
    }
  }, []);

  return null;
};

export default IntercomScript;
```
